### PR TITLE
Pluralize create- and destroy-filesystems

### PR DESCRIPTION
**Pull request by mulkieran**

### CODE REVIEW

**Comment by mulkieran** [comment]
This supercedes #179.

**Comment by mulkieran** [comment]
@agrover @trgill final review? I'm happy for this to go in at any time. Then I will proceed with pluralizing ```remove_devs``` and ```remove_cache_devs```. By doing this, we move action from D-Bus methods into Engine methods, which should simplify the ```GetManagedObjects``` refactoring.

**Comment by trgill** [comment]
:+1: Ok

File: src/engine/engine.rs [hunk 1]
```diff
@@ -84,23 +75,39 @@ impl From<nix::Error> for EngineError {
...
 pub trait Pool: Debug {
-    fn create_filesystem(&mut self,
-                         name: &str,
-                         mount_point: &str,
-                         quota_size: Option<u64>)
-                         -> EngineResult<()>;
-    fn create_snapshot(&mut self, snapshot_name: &str, source: &str) -> EngineResult<()>;
+    /// Creates the filesystems specified by specs.
+    /// Returns a list of the names of filesystems actually created.
+    /// Returns an error if any filesystems of the given name occur.
```
Comment (agrover) [inline src/engine/engine.rs]: s/occur/exist/ please

File: src/engine/engine.rs [hunk 2]
```diff
@@ -84,23 +75,39 @@ impl From<nix::Error> for EngineError {
...
+    fn create_filesystems<'a, 'b, 'c>(&'a mut self,
+                                      mut specs: Vec<(&'b str, &'c str, Option<u64>)>)
```
Comment (agrover) [inline src/engine/engine.rs]: document what specs' tuple members are here?

Comment (agrover) [inline src/engine/engine.rs]: do we need mut in `mut specs` here, or only in the implementation? We're moving ownership, of course it's mut. (really a question about Rust I don't know the answer to)

**Comment by mulkieran** [comment]
@agrover I have added a new commit that changes the type of the ```create_filesystems()``` spec parameter. I do not feel ready to document the specs parameter yet, as we haven't yet very clearly defined how it is used.

**Comment by mulkieran** [comment]
@agrover @trgill merge now?

**Comment by trgill** [comment]
:+1:

**Comment by agrover** [comment]
Because of the last change, you can globally replace all instances of `&vec![` with `&[`. 👍 with that change made.

**Pull request merged**